flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSource.java

```diff
@@ -102,4 +105,154 @@ public void testLimitPushDown() {
     Assert.assertEquals("should have 1 record", 1, mixedResult.size());
     Assert.assertArrayEquals("Should produce the expected records", mixedResult.get(0), new Object[] {1, "a"});
   }
+
+  @Test
+  public void testFilterPushDown() {
+    sql("INSERT INTO %s  VALUES (1,'a'),(2,'b'),(3,CAST(null AS VARCHAR))", TABLE_NAME);
+
+    String expectedExplain = "FilterPushDown";
+
+    // not push down
+    String sqlNoPushDown = "SELECT * FROM " + TABLE_NAME + " WHERE data LIKE '%a%' ";
+    String explainNoPushDown = getTableEnv().explainSql(sqlNoPushDown);
+    assertFalse("explain should not contains FilterPushDown", explainNoPushDown.contains(expectedExplain));
+
+    // equal
+    String sqlLiteralRight = String.format("SELECT * FROM %s WHERE id = 1 ", TABLE_NAME);
+    String explain = getTableEnv().explainSql(sqlLiteralRight);
+    assertTrue("explain should contains FilterPushDown", explain.contains(expectedExplain));
```

Comment:
I will suggest to have a more strict assertion to validate the pushed  filter's internal information, see https://github.com/generic-datalake/iceberg-poc/pull/2/files#diff-5d18d1ff127d1dc70a9a15bbe941f2b6f9d28b3015924f601ac1f722914099dbR96